File: src/main/java/ar/edu/unq/spring/service/interfaces/InventarioService.java
package ar.edu.unq.spring.service.interfaces;

import ar.edu.unq.spring.modelo.Item;

import java.util.Collection;
import java.util.Optional;

public interface InventarioService {
    Optional<Item> getItem(Long itemId);
    Collection<Item> allItems();
    Item heaviestItem();
    Item guardarItem(Item item);
    void recoger(Long personajeId, Long itemId);
    Collection<Item> getMasPesados(int peso);
    Collection<Item> getItemsPersonajesDebiles(int vida);
    void clearAll();
    void deleteItem(Item item);
}

File: src/main/kotlin/ar/edu/unq/spring/service/interfaces/InventarioService.kt
package ar.edu.unq.spring.service.interfaces

import ar.edu.unq.spring.modelo.Item
import ar.edu.unq.spring.modelo.Personaje

/* Eviten el Spanglish :D */
interface InventarioService {

    fun getItem(itemId:Long): Item
    fun allItems(): Collection<Item>
    fun heaviestItem(): Item
    fun guardarItem(item: Item): Item
    fun recoger(personajeId: Long, itemId: Long)
    fun getMasPesados(peso: Int): Collection<Item>
    fun getItemsPersonajesDebiles(vida: Int): Collection<Item>
    fun clearAll()
    fun deleteItem(item: Item)
}